src/ai/openaiProvider.ts

```diff
@@ -12,12 +12,15 @@ const maxCodeCharacters = 12000;
 export class OpenAIProvider implements AIProvider {
 	readonly id = 'openai';
 	readonly name = 'OpenAI';
+	private model: OpenAIChatCompletionModels = 'gpt-3.5-turbo';
 
 	constructor(private readonly container: Container) {}
 
 	dispose() {}
 
 	async generateCommitMessage(diff: string, options?: { context?: string }): Promise<string | undefined> {
+		this.model = configuration.get('experimental.openAIModel') || 'gpt-3.5-turbo';
```

Comment:
Could pull this code into its own method (something like `ensureModel`) which is `(force = false): void`.  

```suggestion
		this.ensureModel();
```

Reply:
Sorry, @d13 ! I focused on something else. I should have followed up sooner.